### PR TITLE
ViewModel-implementation

### PR DESCRIPTION
**Pull request by Farajialireza82**
Using ViewModel to Get SongList and SwapiData

### CODE REVIEW

File: app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java [hunk 1]
```diff
@@ -128,18 +126,128 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
+        songListFragmentViewModel = ViewModelProviders.of(this).get(SongListFragmentViewModel.class);
+
+        songListFragmentViewModel.InitializeProcess();
+
+        songListFragmentViewModel.reciveSongs().observe(this, new Observer<ArrayList<MusicModel>>() {
+
+            @Override
+            public void onChanged(ArrayList<MusicModel> musicModels) {
+
+                recyclerViewAdapter.notifyDataSetChanged();
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java]: inja listi ke az ViewModel miad ro bayad be Adapter bedi ta neshon bede

File: app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java [hunk 2]
```diff
@@ -128,18 +126,128 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
+
+    }
+
+
+    public class AsyncTaskMusicSetter extends android.os.AsyncTask<View, PostMan, RecyclerViewAdapter> {
 
         private WeakReference<MainActivityClass> activityWeakReference;
         private WeakReference<SongListFragment> fragmentWeakReference;
         private WeakReference<RecyclerViewAdapter> recyclerViewAdapterWeakReference;
         RecyclerViewAdapter viewAdapter;
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java]: gharar bod chizi mostaghom to AsyncTask negah dashte nashe
har chi niaze faghat ba WeakReference

File: app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import com.example.galaxyplayer.Retrofit.SwapiClient;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+import androidx.lifecycle.MutableLiveData;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class PeopleRepository {
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java]: afarin ke Repository sakhti

File: app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+    private static PeopleRepository instance;
+    private List<String> dataSet = new ArrayList<>();
+
+    public static PeopleRepository getInstance(){
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java]: chera dari singletone esh mikoni?

File: app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+import java.util.List;
+import androidx.lifecycle.MutableLiveData;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class PeopleRepository {
+
+    private static PeopleRepository instance;
+    private List<String> dataSet = new ArrayList<>();
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java]: behtare bejaye inke list ro bala negah dari 
LiveData ro bezari bala bad har dafe ke niaze yek list e jadid behesh set koni

File: app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    public void askForPeopleInfo(Response<People> response) {
+
+        List<Person> personList = response.body().getResults();
+
+        for (int i = 0; i < personList.size(); i++) {
+
+            dataSet.add(personList.get(i).getName());
+
+
+        }
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java]: inja mitoni vaghti loop tamom shod
ye list jadid besazi
baad list ro set koni be livedata

File: app/src/main/java/com/example/galaxyplayer/repositories/SongListRepo.java [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+        return instance;
+
+    }
+
+    public MutableLiveData<ArrayList<MusicModel>> getMusicData(Application application){
+
+        getSongs(application);
+
+        MutableLiveData <ArrayList<MusicModel>> data = new MutableLiveData<>();
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/repositories/SongListRepo.java]: livedata ye inja ham mitone mese ghabli behtar beshe

File: app/src/main/java/com/example/galaxyplayer/repositories/SongListRepo.java [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+        data.setValue(musicModels);
+        return data;
+
+    }
+
+    private void getSongs(Application application) {
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/repositories/SongListRepo.java]: in felan bad nist
vali khob in hamchenan roye thread e aslie
be nazaram betoni ba asyncTask bebarish roye ye Thread e dige

(albate negaran nabash baadan rahaye kheilii behtari az async task yad migiri)

File: app/src/main/java/com/example/galaxyplayer/repositories/SwapiDataRepository.java [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+        }
+
+        return instance;
+
+    }
+
+    public MutableLiveData<List<String>> getFinalArray() {
+
+        setupArrayList();
+        MutableLiveData<List<String>> data = new MutableLiveData<>();
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/repositories/SwapiDataRepository.java]: in livedata ham hamintor

File: app/src/main/java/com/example/galaxyplayer/repositories/SwapiDataRepository.java [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+import androidx.lifecycle.MutableLiveData;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class SwapiDataRepository {
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/repositories/SwapiDataRepository.java]: in repo fek konam betone ba repo balayi yeki beshe

File: app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+
+    MutableLiveData<ArrayList<MusicModel>> songs;
+
+    public SongListRepo songListRepo = new SongListRepo();
+
+    public void InitializeProcess() {
+
+        if (songs != null) {
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java]: chera hamchin if i vojod dare?

File: app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+import com.example.galaxyplayer.repositories.SongListRepo;
+
+import java.util.ArrayList;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.AndroidViewModel;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+public class SongListFragmentViewModel extends AndroidViewModel {
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java]: estefade az AndroidViewModel ziad tosie nemishe 
vali badan motevajeh mishim chera

File: app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
+    public SongListFragmentViewModel(@NonNull Application application) {
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java]: age mikhay viewmodel i besazi ke to ye constructor ye meghdari ro migire bayad baraye on ViewModel hatman yek ViewModelFactory ham besazi
search kon
(albate AndroidViewModel niazi be Factory nadare)

File: app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+        super(application);
+    }
+
+
+    MutableLiveData<ArrayList<MusicModel>> songs;
+
+    public SongListRepo songListRepo = new SongListRepo();
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java]: behtare in Repo toye constructor sakhte beshe

File: app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java (hunk 1 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java]: Chashm :)

File: app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java (hunk 2 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java]: Az in Dige Aslan Estefadei nadaram

File: app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java (hunk 4 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java]: Vala to Amozesh didam . az singelton esetefa kard manam hamin kar ro kardam?
eshtebah ke nist?

File: app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java (hunk 6 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java]: Anjam Shod!

File: app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java (hunk 3 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java]: 👼

File: app/src/main/java/com/example/galaxyplayer/repositories/SongListRepo.java (hunk 8 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/galaxyplayer/repositories/SongListRepo.java]: Ettefaghan to fekr in bodam ke ba in kar dige amalan az AsyncTask Estefadei nadarim.

File: app/src/main/java/com/example/galaxyplayer/repositories/SwapiDataRepository.java (hunk 10 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/galaxyplayer/repositories/SwapiDataRepository.java]: Nemidonam Vala.

File: app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java (hunk 11 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java]: Nemidonam Chre Hamchin Kari Kardam

File: app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java (hunk 12 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java]: manam Vaghea Emroz Gij shodam
Ishallah ke dige niazi behesh nashe

File: app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java (hunk 13 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java]: Khob Khodaro shokr

File: app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java (hunk 14 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java]: Anjam Shod

File: app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java (hunk 2 above)
Comment (worldsnas) [reply app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java]: pakesh kon bi zahmat!

File: app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java (hunk 4 above)
Comment (worldsnas) [reply app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java]: na nist.
vali khob behtare har vaght niaze ye object jadid sakhte beshe
man pishnahad mikonam intori beri jelo

File: app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java (hunk 6 above)
Comment (worldsnas) [reply app/src/main/java/com/example/galaxyplayer/repositories/PeopleRepository.java]: commit nakardi baz?!

File: app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java (hunk 1 above)
Comment (worldsnas) [reply app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java]: commit nakardi?!

File: app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java (hunk 11 above)
Comment (worldsnas) [reply app/src/main/java/com/example/galaxyplayer/viewModel/SongListFragmentViewModel.java]: fek kon va dalilesho bego!

File: app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java [hunk 15]
```diff
@@ -116,31 +117,36 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
 
         userName = sharedPreferences.getString(NAME, "");
 
         userGreetings.setText("Welcome Dear " + userName);
 
+
         songListFragmentViewModel = ViewModelProviders.of(this).get(SongListFragmentViewModel.class);
 
-        songListFragmentViewModel.InitializeProcess();
+        SongListRepo songListRepo = new SongListRepo();
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java]: in ghalate
repo bayad toye viewModel sakhte she

Comment (worldsnas) [reply app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java]: benazaram behtare baraye viewModel hat Factory besazi

File: app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java [hunk 16]
```diff
@@ -72,7 +72,7 @@
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
 
-        view = inflater.inflate(R.layout.activity_main, container, false);
+        view = inflater.inflate(R.layout.ac, container, false);
```
Comment (worldsnas) [inline app/src/main/java/com/example/galaxyplayer/Fragments/SongListFragment.java]: chera inkaro kardi?
intori khanayish kheili payin omade

**Pull request merged**